### ISSUE TITLE
l10ni18n not loading locale files?

### ISSUE THREAD

**Issue by devanonyme** [WebDevStudios/CMB2#711]
I was expecting the data to be displayed in french (my wordpress config) but it was displaying in english.

It looks like the files themselves are not loaded. While stepping it, I see that load_theme_textdomain is called (loaded = true) and the last scenario is not run.

I'm still new to Wordpress but from what I understand them langages file refer to core domain. I'm not sure why the plugin try to get the language file there?

Maybe the last load should be done all the time?

Someone could give me an idea of the load scenario and what cases it shoud cover?

$loaded = load_plugin_textdomain( 'cmb2', false, '/languages/' );

			if ( ! $loaded ) {
				$loaded = load_muplugin_textdomain( 'cmb2', '/languages/' );
			}

			if ( ! $loaded ) {
				$loaded = load_theme_textdomain( 'cmb2', get_stylesheet_directory() . '/languages/' );
			}

			if ( ! $loaded ) {
				$locale = apply_filters( 'plugin_locale', get_locale(), 'cmb2' );
				$mofile = dirname( __FILE__ ) . '/languages/cmb2-' . $locale . '.mo';
				load_textdomain( 'cmb2', $mofile );
			}

**Comment by tw2113** [comment]
At least for me, it was reaching that last `load_textdomain( 'cmb2', $mofile );` line, but it was also finding a mo file that matched and loaded it.

I got this error for a bug I had recently been debugging:

```
Configuration de Metabox requise pour l'obtention d'un paramètre ID`
```
One thing about the .po/.mo files is that majority of the content in there is for the example code. It'd get used when copy/pasting as is. At the exact same time, providing the translated string at the time of your own use of the field would also translate it, so the .po/.mo version isn't as important at that point. There definitely are areas that wouldn't be immediately translated by the dev, and those are more in the dropdown options for things like date/time etc.

**Comment by tw2113** [comment]
any changes with this one on your end @devanonyme ?

**Comment by devanonyme** [comment]
Sorry, no. I had to put that aside as we are on a agressive delivery timeline.

I was considering CMB2 for our project but being new to wordpress and also the organisation so I missed leverage to "sell" it. We met a few expert that we all about ACF and he Wordpress expert that we hired tought that it was simpler to reimplement from scratch considering the constraint that we had (by ex. responsive and also accessibility support).

I would personnaly have prefered creating new fields in cmb2 and even managed a custom fork to that have more control on the html output than a whole new developpement (I still think there will be alot of plumbing to do /redo)!

Except for that bug, CMB2 was looking usable so I know I could still be reverting to this solution if we ever have to reconsider CMB2.

**Comment by tw2113** [comment]
Noted, thanks.

I'm having a hard time calling this a bug, or at least if it is, it was one I wasn't managing to recreate. My hunches lean towards it being something specific the site originally tried on. However, I'll leave it open for the meantime, in case someone else comes in with more information.

**Comment by devanonyme** [comment]
Yes I understand, there should be more details / context, I would probably close is as "not able to reproduce" if I had to deal wich such bug! :)

I was using 4.5.3 + 'responsive' theme on a dev setup using varying vagrant (on Windows 7) and "fr_ca" as lang config in wordpress, so nothing really fancy.

I noticed the files where not loading because the form had some field element displaying text in english and this where I started to debug it to see where the translation where load.
